src/msdfgl.c

```diff
@@ -1022,6 +1022,17 @@ float msdfgl_printf(float x, float y, msdfgl_font_t font, float size, int32_t co
         return x;
     }
 
+    // Used for newline support
+    size_t y_init = y;
+    size_t x_init = x;
+    unsigned int n_newlines = 0;
+
+    float advance;
+    if (flags & MSDFGL_VERTICAL)
+        advance = (font->vertical_advance * (size * font->context->dpi[0] / 72.0f) / font->face->units_per_EM);
+    else
+        advance = (font->vertical_advance * (size * font->context->dpi[1] / 72.0f) / font->face->units_per_EM);
```

Comment:
Should this be `font->horizontal_advance` instead?

Reply:
so, i'm not sure what's the best thing here. the vertical_advance is calculated in `_msdfgl_init_font_internal` as `f->vertical_advance = (float)(f->face->ascender - f->face->descender);`, but clearly cannot be the same for horizontal advance. `font` is a `FT_Face`. Here is a relevant bit from free type (2.13.3):

```c
typedef struct  FT_FaceRec_
  {
    // ...

    /* The following member variables (down to `underline_thickness`) */
    /* are only relevant to scalable outlines; cf. @FT_Bitmap_Size    */
    /* for bitmap fonts.                                              */
    FT_BBox           bbox;

    FT_UShort         units_per_EM;
    FT_Short          ascender;
    FT_Short          descender;
    FT_Short          height;

    FT_Short          max_advance_width;
    FT_Short          max_advance_height;

    FT_Short          underline_position;
    FT_Short          underline_thickness;

    FT_GlyphSlot      glyph;
    FT_Size           size;
    FT_CharMap        charmap;

    // ...
  } FT_FaceRec;
```

I've redacted some irrelevant fields there. I guess we could do something like `f->horizontal_advance = (float)(f->face->max_advance_width);` but that relies on the font to have a correctly specified horizontal advance.

I guess what i've done is basically using the vertical advance as a proxy for 'how much apart should any two lines of text be?'. At the very least I think I should add a comment to clarify that. Also happy to implement the `max_advance_width` or any other suggestion (`units_per_EM` aka the width of the letter 'm'? unsure whether that's better.)

Reply:
Using the font size, and `line_spacing` as a multiplier to it does sound similar to how other tools seem to be doing it.

Let's go with the simple solution now, and if a problematic font comes up we can revisit.

Reply:
to clarify: simple solution aka how it is now, or `size * font->context->dpi[...] / 72.0f`, so effectively fixed line spacing of 1.0?

Reply:
Yes, that sounds good to me 👍🏼